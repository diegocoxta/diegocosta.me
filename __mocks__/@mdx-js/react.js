module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
