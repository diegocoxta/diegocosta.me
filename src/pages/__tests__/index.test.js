import React from 'react';
import { render } from '@testing-library/react';

jest.mock('../../components/Header', () => () => <p>Header</p>);
jest.mock('../../components/Footer', () => () => <p>Footer</p>);
jest.mock('../../components/Metatags', () => () => <p>Metatags</p>);
jest.mock('../../components/Search', () => () => <p>Search</p>);

import IndexPage from '../index';

describe('<IndexPage>', () => {
  it('renders properly', () => {
    const data = {
      articles: {
        edges: [
          {
            node: {
              frontmatter: {
                title: 'Title 1',
                date: '2021-01-16T22:12:03.284Z',
                tags: [],
                description: 'Awesome first article',
              },
              fields: {
                slug: '/path-to-article',
                readingTime: {
                  minutes: 5,
                },
              },
              excerpt: 'Awesome first article',
            },
          },
          {
            node: {
              frontmatter: {
                title: 'Title 2',
                date: '2020-01-18T22:12:03.284Z',
                tags: [],
                description: 'Awesome second article',
              },
              fields: {
                slug: '/path-to-article-2',
                readingTime: {
                  minutes: 2,
                },
              },
              excerpt: 'Awesome first article',
            },
          },
        ],
      },
      articlesMdx: {
        edges: [],
      },
    };

    const { baseElement, getAllByTestId, getByText } = render(<IndexPage data={data} />);
    expect(getAllByTestId('article-item').length).toEqual(2);
    expect(getByText('Awesome second article')).toBeTruthy();
    expect(getByText('Awesome first article')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
