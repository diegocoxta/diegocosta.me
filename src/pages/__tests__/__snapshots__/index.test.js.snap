// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexPage> renders properly 1`] = `
.c11 {
  color: #2F3D4F;
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c11:hover,
.c11:focus {
  border-bottom: 1px solid #2F3D4F;
  outline: none;
}

.c12 {
  line-height: 1.5;
  font-size: 19px;
  margin-top: 10px;
}

.c12 a {
  color: #3A3D7D;
  text-decoration: none;
  border-bottom: 1px solid #FFFFFF;
}

.c12 a:hover,
.c12 a:focus {
  border-bottom: 1px solid #3A3D7D;
  outline: none;
}

.c12 code {
  background: #3A3D7D40;
  padding: 2px 6px;
  border-radius: 10px;
  font-size: 16px;
}

.c12 blockquote {
  border-left: 5px solid #3A3D7D;
  padding-left: 20px;
  margin-left: 0;
}

.c12 img {
  box-shadow: none!important;
}

.c12 h3 {
  font-size: 28px;
}

.c12 table.uses-table {
  width: 100%;
  margin: 40px 0;
}

.c12 table.uses-table td {
  text-align: center;
}

.c0 {
  max-width: 960px;
  margin: 0 auto;
  padding: 0 20px;
}

.c10 {
  margin: 20px auto 60px auto;
  font-weight: 400;
  color: #2F3D4F;
}

.c2 {
  text-decoration: none;
  box-shadow: none;
  display: flex;
  align-items: center;
}

.c2:focus {
  outline: none;
}

.c3 {
  font-size: 15vw;
  color: #2F3D4F;
  font-weight: 700;
  text-transform: lowercase;
  margin: 0;
}

.c4 {
  color: #3A3D7D;
}

.c8 {
  background: transparent;
  border: 0px;
  padding: 5px;
  border-radius: 50px;
  width: 40px;
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  color: #2F3D4F;
  transition: background 0.3s linear;
  cursor: pointer;
}

.c8:focus {
  outline: none;
}

.c8:hover {
  background: #2F3D4F1A;
}

.c6 {
  background: transparent;
  border: 2px solid #3A3D7D;
  border-radius: 18px;
  height: 34px;
  width: 50px;
  padding: 0px;
  display: flex;
  justify-content: flex-end;
  transition: all 1s linear;
  margin: 0 20px;
  cursor: pointer;
}

.c6:focus {
  outline: none;
}

.c7 {
  height: 30px;
  width: 30px;
  background: #3A3D7D;
  border-radius: 20px;
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
}

.c1 {
  margin: 16px auto 40px auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.c5 {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.c9 {
  background: linear-gradient(90deg, #FFFFFF 20px, transparent 1%) center,linear-gradient(#FFFFFF 20px, transparent 1%) center,#2F3D4F;
  background-size: 22px 22px;
  height: 120px;
  margin: 40px 0;
  width: 100%;
}

@media (min-width: 760px) {
  .c3 {
    font-size: 110px;
  }
}

@media (min-width: 760px) {
  .c1 {
    margin: 40px auto 40px auto;
  }
}

<body>
  <div>
    <section
      class="c0 c1"
    >
      <a
        class="c2"
        data-testid="logo-link"
        href="/"
      >
        <h1
          class="c3"
        >
          Diego
          <span
            class="c4"
            data-testid="logo-lastname"
          >
            C
            .
          </span>
        </h1>
      </a>
      <div
        class="c5"
      >
        <button
          aria-label="Change color scheme"
          class="c6"
        >
          <div
            class="c7"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 11a3 3 0 1 1 0-6 3 3 0 0 1 0 6m0 1a4 4 0 1 0 0-8 4 4 0 0 0 0 8M8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0m0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13m8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5M3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8m10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0m-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0m9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707M4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708"
              />
            </svg>
          </div>
        </button>
        <button
          class="c8"
        >
          <svg
            fill="currentColor"
            height="28"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.5 2A1.5 1.5 0 0 1 5 3.5V5H3.5a1.5 1.5 0 1 1 0-3M6 5V3.5A2.5 2.5 0 1 0 3.5 6H5v4H3.5A2.5 2.5 0 1 0 6 12.5V11h4v1.5a2.5 2.5 0 1 0 2.5-2.5H11V6h1.5A2.5 2.5 0 1 0 10 3.5V5zm4 1v4H6V6zm1-1V3.5A1.5 1.5 0 1 1 12.5 5zm0 6h1.5a1.5 1.5 0 1 1-1.5 1.5zm-6 0v1.5A1.5 1.5 0 1 1 3.5 11z"
            />
          </svg>
        </button>
      </div>
    </section>
    <section
      class="c0"
    />
    <div
      class="c9"
    />
    <section
      class="c0 c10"
      data-testid="article-item"
    >
      <a
        class="c11"
        data-testid="title"
        href="/path-to-article"
      >
        Title 1
      </a>
      <section
        class="c12"
        html="Awesome first article"
      >
        Awesome first article
      </section>
    </section>
    <section
      class="c0 c10"
      data-testid="article-item"
    >
      <a
        class="c11"
        data-testid="title"
        href="/path-to-article-2"
      >
        Title 2
      </a>
      <section
        class="c12"
        html="Awesome second article"
      >
        Awesome second article
      </section>
    </section>
    <p>
      Footer
    </p>
  </div>
</body>
`;
